File: 283-move-zeroes/move-zeroes.go
func moveZeroes(nums []int)  {
    nonZeroPtr := 0
    for i, _ := range nums {
        if nums[i] != 0 {
            nums[nonZeroPtr] = nums[i]
            nonZeroPtr++
        }
    } 

    for i := nonZeroPtr; i < len(nums); i++ {
        nums[i] = 0
    }
}

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nonZeroPtr = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[nonZeroPtr] = nums[i]
                nonZeroPtr += 1

        for i in range(nonZeroPtr, len(nums)):
            nums[i] = 0
        